src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapVectorTransferFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.opensearch.knn.index.VectorDataType;
+
+public final class OffHeapVectorTransferFactory {
+
+    private OffHeapVectorTransferFactory() {}
+
+    public static <T> OffHeapVectorTransfer<T> getVectorTransfer(final VectorDataType vectorDataType, final int transferLimit) {
+        switch (vectorDataType) {
+            case FLOAT:
+                return (OffHeapVectorTransfer<T>) new OffHeapFloatVectorTransfer(transferLimit);
+            case BINARY:
+                // TODO: Add binary here
+            case BYTE:
+                return (OffHeapVectorTransfer<T>) new OffHeapByteVectorTransfer(transferLimit);
+            default:
+                throw new IllegalArgumentException("Unsupported vector data type: " + vectorDataType);
```

Comment:
"No valid vector transfer implementation" is an implementation detail and doesn't make sense for the user.

Reply:
> "No valid vector transfer implementation" is an implementation detail and doesn't make sense for the user.

The exceptions from these layer doesn't propagate to users. This will be added in logs which will be used by admin and operators. Just saying `Unsupported vector data type` gives no clue what the exception and what are the valid vector datatypes here.